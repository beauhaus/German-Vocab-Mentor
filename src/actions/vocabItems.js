import uuid from 'uuid';



// ADD VOCAB
export const addVocabItem = ({
    word_de = '',
    word_en = '',
    example_de = '',
    example_en = '',
    isShowing = '',
    note = '',
    createdAt = 0 } = {}
) => ({
    type: 'ADD_VOCAB_ITEM',
    vocab: {
        "id": uuid(),
        word_de,
        word_en,
        example_de,
        example_en,
        isShowing,
        note,
        createdAt
    }
})

// REMOVE VOCAB

export const removeVocabItem = ({ id } = {}) => ({
    type: 'REMOVE_VOCAB_ITEM',
    id
});

// EDIT VOCAB
export const editVocabItem = (id, updates) => ({
    type: 'EDIT_VOCAB_ITEM',
    id,
    updates
});