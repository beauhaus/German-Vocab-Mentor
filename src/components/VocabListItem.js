import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const VocabListItem = ({ dispatch, id, note, difficulty, isShowing, word_de, word_en, example_en, example_de, createdAt }) => {

createdAt = moment(createdAt).format("M•D•YY ddd h:mma")

const slicedDateStr = "test";
       return (
              <div className="temp-voc-item">
                     <Link to={`/edit${id}`}>
                            <h1 className="vocab-edit-link">&#x270E;</h1>
                     </Link>
                     <h2>{word_de} – {word_en} <span>{isShowing ? "👁" : "🚫"}</span></h2>
                     <h2>{example_de}</h2>
                     <h2>{example_en}</h2>
                     <h2>Note: {note}</h2>
                     <h2>Difficulty: {difficulty}</h2>
                     <h2>{createdAt}</h2>
                     <br /><br />
                     <hr />
                     <br />
              </div>
       )
}

export default VocabListItem;