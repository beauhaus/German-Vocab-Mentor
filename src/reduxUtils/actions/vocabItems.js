import uuid from 'uuid';

export const addVocabItem = ({ word_de = '', word_en = '', note = '', difficulty = 10, isShowing = true, example_de = '', example_en = '', createdAt = 0 } = {}) => ({
    type: 'ADD_VOCAB_ITEM',
    vocabItem: {
        _id: uuid(),
        word_de,
        word_en,
        note,
        example_de,
        example_en,
        difficulty,
        createdAt,
        isShowing
    }
})

export const removeVocabItem = ({ _id } = {}) => ({
    type: 'REMOVE_VOCAB_ITEM',
    _id
})

export const editVocabItem = (_id, updates) => ({
    type: 'EDIT_VOCAB_ITEM',
    _id,
    updates
})