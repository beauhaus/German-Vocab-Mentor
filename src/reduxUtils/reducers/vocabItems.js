const vocabItemsReducerDefaultState = [];

const vocabItemsReducer = (state = vocabItemsReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_VOCAB_ITEM':
            return [...state, action.vocabItem]
        case 'REMOVE_VOCAB_ITEM':
            return state.filter(({ _id }) => _id !== action._id)
        case 'EDIT_VOCAB_ITEM':
            return state.map((vocabItem) => {
                if (vocabItem._id === action._id) {
                    return {
                        ...vocabItem,
                        ...action.updates
                    }
                } else {
                    return vocabItem;
                }
            })
        default:
            return state;
    }
}

export default vocabItemsReducer;