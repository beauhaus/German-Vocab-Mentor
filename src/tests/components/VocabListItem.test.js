import React from 'react';
import {shallow} from 'enzyme';
import vocabItems from '../fixtures/vocabItems';

import VocabListItem from '../../components/VocabListItem';

test('should render vocab list item correctly', () => {
    const wrapper = shallow(<VocabListItem {...vocabItems[0]}/>)
    expect(wrapper).toMatchSnapshot();
})
