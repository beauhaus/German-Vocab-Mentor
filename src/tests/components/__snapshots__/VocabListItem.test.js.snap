// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render vocab list item correctly 1`] = `
<div
  className="temp-voc-item"
>
  <Link
    replace={false}
    to="/edit1"
  >
    <h1
      className="vocab-edit-link"
    >
      ‚úé
    </h1>
  </Link>
  <h2>
    Erinnerungen
     ‚Äì 
    Memories
     
    <span>
      üëÅ
    </span>
  </h2>
  <h2>
    es weckt Erinnerungen
  </h2>
  <h2>
    It awakens (brings back) memories
  </h2>
  <h2>
    Note: 
    memory note
  </h2>
  <h2>
    Difficulty: 
    8
  </h2>
  <h2>
    12‚Ä¢31‚Ä¢69 Wed 7:00pm
  </h2>
  <br />
  <br />
  <hr />
  <br />
</div>
`;
